### PR TITLE
Updated GetNOTAM.py

### PR DESCRIPTION
**Pull request by BrendenDeWitt**
Added if statements for the start and end date so it can work whether they are inputted or not.

### CODE REVIEW

**Review by Gabrieldowen — APPROVED** [review]
this works! I am not sure what its returning when you dont enter a date it seems to be a big random mix with dates going back to atleast 2021. For this reason in the future we probably wanna give the user a limited date range

File: GetNOTAM.py [hunk 1]
```diff
@@ -13,15 +13,18 @@
 #getNotam: takes the lat, long, start and end time and the page number then runs an API call to the FAA for a json of the api
 #@returns parsed_req: the json of the request
 def getNotam(effectiveStartDate, effectiveEndDate, longitude, latitude, pageNum, radius):
-  # Convert datetime strings to the desired format
-  effectiveStartDate = datetime.strptime(effectiveStartDate, "%Y-%m-%d %H:%M:%S").strftime("%Y-%m-%dT%H:%M:%SZ")
-  effectiveEndDate = datetime.strptime(effectiveEndDate, "%Y-%m-%d %H:%M:%S").strftime("%Y-%m-%dT%H:%M:%SZ")  
+  # Convert start and end dates to the desired format if provided
+  if effectiveStartDate:
```
Comment (n-grounds) [inline GetNOTAM.py:17]: if `not effectiveStartDate` what will happen?  does the API call below still work?  if not, we should raise an Exception of some kind to indicate the function was called with invalid parameter values

File: GetNOTAM.py [hunk 2]
```diff
@@ -13,15 +13,18 @@
...
+        
+  if effectiveEndDate:
+    effectiveEndDate = datetime.strptime(effectiveEndDate, "%Y-%m-%d %H:%M:%S").strftime("%Y-%m-%dT%H:%M:%SZ") 
   
   url = 'https://external-api.faa.gov/notamapi/v1/notams'
   url = (f"{url}?responseFormat=geoJson&effectiveStartDate={effectiveStartDate}"
        f"&effectiveEndDate={effectiveEndDate}&locationLongitude={longitude}"
        f"&locationLatitude={latitude}&locationRadius={radius}"
-       f"&pageNum={pageNum}&pageSize=50"
+       f"&pageNum={pageNum}&pageSize=1000"
```
Comment (n-grounds) [inline GetNOTAM.py:27]: I think there is an assumption that requesting fewer but larger pages is faster, or is this verified?  is there an issue/ticket in the backlog to maybe test that assumption or figure out if there is an optimal page size?

File: GetNOTAM.py (hunk 1 above)
Comment (BrendenDeWitt) [reply GetNOTAM.py:17]: The API call works without a startDate or endDate so if not the program runs as normal. If a startDate and endDate are inputted then they need to be converted to the required format.

**Pull request merged**